doc/chromebooks/setting_gbb_flags.md

```diff
@@ -0,0 +1,200 @@
+
+# Setting gbb flags
+
+gbb flags make it so if your chromebook battery runs dry, you won't lose access to your system
+
+in order to set gbb flags you need to [enable developermode](./preparing-chromebook.md) and disable hardware write protection
+
+1. start by booting chrome os and entering tty with ```[ctrl]``` ```[alt]``` ```[->]``` or ```[ctrl]``` ```[alt]``` ```[refresh]``` or ```[ctrl]``` ```[alt]``` ```[f2]```
+
+2. login with username chronos (no password required)
+
+3. move to directory where ```touch testfile``` works (```cd /tmp``` should be ok)
+
+4. check software write protection with ```flashrom --wp-status```
+it will probaly enable, you can disable it with ```flashrom --wp-disable``` and check it again with ```flashrom --wp-status```
+
+*Note. this will only work if the hardware write protection has been disabled beforehand*
+
+<details>
+<summary>Recommended step</summary>
+
+read the firmware from the flash into a file ```bios.bin``` via the command ```flashrom -r bios.bin```.
+
+it is a good idea to copy this file to a safe place outside of the chromebook now (sd card, usb stick etc.) to have a copy of the original unmodified firmware around just in case ...
+</details>
+
+*Note. in case the boot screen should be changed as well, [look here]() it might be a good idea to do this first to avoid resetting the changed gbb settings again then*
+
+<details>
+<summary>there are a few diffrent gbb flags</summary>
+
+```
+  GBB_FLAG_DEV_SCREEN_SHORT_DELAY     0x00000001
+  GBB_FLAG_LOAD_OPTION_ROMS           0x00000002
+  GBB_FLAG_ENABLE_ALTERNATE_OS        0x00000004
+  GBB_FLAG_FORCE_DEV_SWITCH_ON        0x00000008
+  GBB_FLAG_FORCE_DEV_BOOT_USB         0x00000010
+  GBB_FLAG_DISABLE_FW_ROLLBACK_CHECK  0x00000020
+  GBB_FLAG_ENTER_TRIGGERS_TONORM      0x00000040
+  GBB_FLAG_FORCE_DEV_BOOT_LEGACY      0x00000080
+  GBB_FLAG_FAFT_KEY_OVERIDE           0x00000100
+  GBB_FLAG_DISABLE_EC_SOFTWARE_SYNC   0x00000200
+  GBB_FLAG_DEFAULT_DEV_BOOT_LEGACY    0x00000400
+  GBB_FLAG_DISABLE_PD_SOFTWARE_SYNC   0x00000800
+  likely more
+```
+*Note. some values may be not supported by all systems*
+</details>
+
+<details>
+<summary>it's recommended to use 0x19 for gbb flags</summary>
+
+which combines
+
+- GBB_FLAG_DEV_SCREEN_SHORT_DELAY 0x00000001 - initial boot screen only for 2 seconds instead of the default 30 seconds and no beep afterwards
+- GBB_FLAG_FORCE_DEV_SWITCH_ON 0x00000008 - keep developer mode enabled by default
+- GBB_FLAG_FORCE_DEV_BOOT_USB 0x00000010 - keep the possibility to boot from usb/sd card enabled by default
+</details>
+
+5. set the desired gbb flags via the command ```/usr/share/vboot/bin/set_gbb_flags.sh 0x19```,
+on newer chromeos versions (around 112+ or so) one should use ```futility gbb --set --flash --flags=0x19``` instead (but the old command from above seems to still work as well)
```

Comment:
on newest chromeos versions the old command no longer works - will have to fix this in my arm chromebook repo as well
```suggestion
on newer chromeos versions (around 112+ or so) one should use ```futility gbb --set --flash --flags=0x19``` instead
```